### PR TITLE
Added the isometric legacy map importer

### LINKED ISSUE #9335: Legacy isometric map import
Work in progress :construction: at

* https://github.com/pchote/OpenRA/tree/ts-map-importer
* ~~https://github.com/Mailaender/OpenRA/tree/ts-map-import~~

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26652688-legacy-isometric-map-import?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Mailaender**
This is a rebased and slightly improved version of the one published by @pchote. It adds argument validation, fixed the file not found errors, uncommented decorational actors in the translation table and made sure it will also work with Red Alert 2 maps. I am submitting this now, because it is quite hard to maintain outside this repository.

Closes https://github.com/OpenRA/OpenRA/issues/9335.

### CODE REVIEW

**Comment by pchote** [comment]
My version of the importer has issues with the map border calculation, and with actor positions being wrong by a couple of cells (possibly due to differences in the corner that TS used as the reference point).  I did not want to merge this until after those issues (especially the second one) had been fixed, because they generate bogus maps.

File: OpenRA.Game/FileFormats/Format80.cs [hunk 1]
```diff
@@ -60,6 +60,70 @@ static void ReplicatePrevious(byte[] dest, int destIndex, int srcIndex, int coun
 			}
 		}
 
+		public static unsafe uint DecodeInto(byte* src, byte* dest) {
```
Comment (pchote) [inline OpenRA.Game/FileFormats/Format80.cs]: This method does not follow our coding conventions.

File: OpenRA.Mods.TS/UtilityCommands/ImportIsometricLegacyMapCommand.cs [hunk 2]
```diff
@@ -0,0 +1,476 @@
...
+using System.IO;
+using System.Linq;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.FileFormats;
+using OpenRA.FileSystem;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.TS.UtilityCommands
+{
+	public class Format5
```
Comment (pchote) [inline OpenRA.Mods.TS/UtilityCommands/ImportIsometricLegacyMapCommand.cs]: This class does not follow our coding conventions, and probably doesn't want to live here.

File: OpenRA.Mods.TS/UtilityCommands/ImportIsometricLegacyMapCommand.cs [hunk 3]
```diff
@@ -0,0 +1,476 @@
...
+		}
+
+		void ReadActors(Map map, IniFile file, string type)
+		{
+			var structuresSection = file.GetSection(type, true);
+			foreach (var kv in structuresSection)
+			{
+				var entries = kv.Value.Split(',');
+
+				//var owner = entries[0];
```
Comment (pchote) [inline OpenRA.Mods.TS/UtilityCommands/ImportIsometricLegacyMapCommand.cs]: dead code

File: OpenRA.Mods.TS/UtilityCommands/ImportIsometricLegacyMapCommand.cs [hunk 4]
```diff
@@ -0,0 +1,476 @@
...
+					float.TryParse(kv.Value, NumberStyles.Float, NumberFormatInfo.InvariantInfo, out ambient);
+					break;
+				default:
+					Console.WriteLine("Ignoring unknown lighting type: `{0}`".F(kv.Key));
+					break;
+				}
+			}
+
+			var node = new MiniYamlNode("World", new MiniYaml("", new List<MiniYamlNode>()
+				{
```
Comment (pchote) [inline OpenRA.Mods.TS/UtilityCommands/ImportIsometricLegacyMapCommand.cs]: This should be unindented by one level.

**Comment by Mailaender** [comment]
Actor position looks fine when I compare houses to streets and bridge huts to bridges:

image

File: OpenRA.Mods.TS/UtilityCommands/MiniLZO.cs [hunk 5]
```diff
@@ -0,0 +1,446 @@
+
+#region Copyright notice
```
Comment (pchote) [inline OpenRA.Mods.TS/UtilityCommands/MiniLZO.cs]: This file does not conform to our project style, and should be rewritten.

**Comment by pchote** [comment]
Basically, this was a proof of concept / prototype that lifted code from other projects without regard to integrating it with our own code base, proper documentation, licenses, or attribution.  It should not be merged until the low level bits have been properly integrated with our own format / parsing code.

**Comment by Phrohdoh** [comment]
@pchote So are we going to consider merging this at all if that is the case?

**Comment by pchote** [comment]
There is a bit of work to address my complaints, but it's by no means impossible.  The only tricky bit is rewriting the MiniLZO implementation, but we are only using one (~150 line) function from it so even that isn't too bad.

**Comment by Mailaender** [comment]
Superseded by https://github.com/OpenRA/OpenRA/pull/10397/ which includes StyleCop cleanups, map bound fixes and doesn't add redundant global lighting variables.

**Pull request closed without merging**